### PR TITLE
feat: toast messages for sticker and ens trxs

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

File: src/app/chat/event_handling.nim [hunk 1]
```diff
@@ -77,11 +77,9 @@ proc handleChatEvents(self: ChatController) =
 
   self.status.events.on(PendingTransactionType.BuyStickerPack.confirmed) do(e: Args):
     var tx = TransactionMinedArgs(e)
+    self.view.transactionCompleted(tx.success, tx.transactionHash)
```
Comment (jrainville) [inline src/app/chat/event_handling.nim:80]: No big deal, but we can pass the `revertReason` already even if we don't listen to it in QML yet

File: ui/app/AppLayouts/Chat/components/StickerPackPurchaseModal.qml [hunk 2]
```diff
@@ -225,6 +225,31 @@ ModalPopup {
...
+            onTransactionWasSent: {
+                toastMessage.title = qsTr("Transaction pending...")
+                toastMessage.source = "../../../img/loading.svg"
+                toastMessage.iconColor = Style.current.primary
+                toastMessage.iconRotates = true
+                toastMessage.link = `${walletModel.etherscanLink}/${txResult}`
+                toastMessage.open()
+            }
+            onTransactionCompleted: {
+                toastMessage.title = !success ? qsTr("Could not buy Stickerpack"): qsTr("Stickerpack bought successfully");
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/components/StickerPackPurchaseModal.qml:240]: Put the qsTr on two lines please

File: ui/app/AppLayouts/Profile/Sections/EnsContainer.qml [hunk 3]
```diff
@@ -273,4 +273,37 @@ Item {
...
+            toastMessage.link = `${walletModel.etherscanLink}/${txResult}`
+            toastMessage.open()
+        }
+        onTransactionCompleted: {
+            switch(trxType){
+                case "RegisterENS":
+                    toastMessage.title = !success ? qsTr("ENS Registration failed"): qsTr("ENS Registration completed");
+                    break;
+                case "SetPubKey":
+                    toastMessage.title = !success ? qsTr("Updating ENS pubkey failed"): qsTr("Updating ENS pubkey completed");
```
Comment (jrainville) [inline ui/app/AppLayouts/Profile/Sections/EnsContainer.qml:293]: qsTr on two lines for the two lines above

**Review by jrainville — COMMENTED** [review]
Good job.
Seeing the ToastMessage being used in 3 different places now makes me realize that we could optimize it more if we wanted since they seem to all use the same icons, colors, etc, but it's no big deal for now

**Review by jrainville — APPROVED** [review]

File: ui/app/AppLayouts/Chat/ChatColumn/ChatComponents/SignTransactionModal.qml [hunk 4]
```diff
@@ -59,7 +59,7 @@ ModalPopup {
             return sendingError.open()
         }
 
-        sendingSuccess.text = qsTr("Transaction sent to the blockchain. You can watch the progress on Etherscan: %2/%1").arg(response.result).arg(walletModel.etherscanLink)
+        sendingSuccess.text = qsTr("Transaction sent to the blockchain. You can watch the progress on Etherscan: %2%1").arg(response.result).arg(walletModel.etherscanLink)
```
Comment (emizzle) [inline ui/app/AppLayouts/Chat/ChatColumn/ChatComponents/SignTransactionModal.qml:62]: Do we still need to show this? I think it was meant to be used as a temporary solution while we implemented the toast messages.

File: ui/app/AppLayouts/Chat/components/StickerPackPurchaseModal.qml [hunk 5]
```diff
@@ -52,7 +52,7 @@ ModalPopup {
             return sendingError.open()
         }
 
-        sendingSuccess.text = qsTr("Transaction sent to the blockchain. You can watch the progress on Etherscan: %2/%1").arg(response.result).arg(walletModel.etherscanLink)
+        sendingSuccess.text = qsTr("Transaction sent to the blockchain. You can watch the progress on Etherscan: %2%1").arg(response.result).arg(walletModel.etherscanLink)
```
Comment (emizzle) [inline ui/app/AppLayouts/Chat/components/StickerPackPurchaseModal.qml:55]: Same as above.

**Review by emizzle — APPROVED** [review]
Really awesome!! 💪

Only comment would be that we likely don't need `MessageDialog`s after txs anymore.

**Pull request merged**